lib/code-caller/index.js

```diff
@@ -234,7 +244,10 @@ module.exports = {
     if (overallErr) {
       throw overallErr;
     } else {
-      return fnResult;
+      // TypeScript doesn't understand our error-handling logic above. If
+      // `overallErr` is falsy, `fnResult` will indeed have type `T`. We'll
+      // cast it to `T` to appease TypeScript.
+      return /** @type {T} */ (fnResult);
```

Comment:
I'm once again tempted to reach for something like https://dev.to/karanpratapsingh/go-like-error-handling-in-typescript-4plm 😅

Reply:
Yuck, but I have no better ideas.